### DISCUSSION TITLE
Load workout @ Google Chrome / Android

### DISCUSSION THREAD

**Discussion by vegancorr** [dvmarinoff/Auuki#123] [Q&A] [answered]
Should the load workout work on Chrome / Android? After selecting the file to upload, it doesn't show the filename on the page.

 It does work on desktop.

**Comment by dvmarinoff** [comment]
> Should the load workout work on Chrome / Android? After selecting the file to upload, it doesn't show the filename on the page.
> 
> It does work on desktop.

Thanks for the report! Will look into it ...

**Comment by dvmarinoff** [comment]
I can't recreate this issue on Android/MacOS + Chrome. Both courses and workouts load with names next to the Load Button and everywhere else. Do you still experience this issue?

**Reply by vegancorr** [reply]
Yes, the issue is still present on Chrome 103 / Android 11 (Xiaomi Mi9 / 12.5.1) @ flux-three.vercel.app / flux-devel.vercel.app ("No file chosen"). The same file loads on Chrome / Ubuntu.

**Reply by vegancorr** [reply]
Funny thing, the same happens on Firefox (I know it's not supported). It loads on Desktop, but not on Android. This makes me believe there is some weird thing happening with Xiaomi's "improvements".

**Reply by vegancorr** [reply]
After further research, it seems it works only with "MIUI optimizations" option enabled in Developer Options. So it's not a bug, the issue is on Xiaomi side.

**Reply by dvmarinoff** [reply]
I am also on Chrome 103, but Android 10, on Samsung Galaxy S9 and S10.

It works for me with Chrome, Android Browser, Samsung Internet, Opera and Firefox.
The code is plain html and vanilla javascript, super simple, no fancy stuff.
Did some quick search for known issues with input type file and Xiaomi devices, but nothing relevant came out.
Will try again tomorrow.

**Reply by dvmarinoff** [reply]
> After further research, it seems it works only with "MIUI optimizations" option enabled in Developer Options. So it's not a bug, the issue is on Xiaomi side.

Just saw that. Nice work!

**Comment by dvmarinoff** [comment, marked as answer]
> After further research, it seems it works only with "MIUI optimizations" option enabled in Developer Options. So it's not a bug, the issue is on Xiaomi side.

Marking this comment as an answer.
